velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -38,6 +38,46 @@ namespace facebook::velox::connector::hive {
 
 namespace {
 
+RowVectorPtr makeRowVector(
+    std::vector<std::string> childNames,
+    const std::vector<VectorPtr>& children,
+    int32_t numRows,
+    velox::memory::MemoryPool* pool) {
+  std::vector<std::shared_ptr<const Type>> childTypes;
+  childTypes.resize(children.size());
+  for (int i = 0; i < children.size(); i++) {
+    childTypes[i] = children[i]->type();
+  }
+  auto rowType = ROW(std::move(childNames), std::move(childTypes));
+  return std::make_shared<RowVector>(
+      pool, rowType, BufferPtr(nullptr), numRows, std::move(children));
```

Comment:
For the 3rd param, it should be better to be input->nulls().

Here I find a good example of how to make a RowVector from a subset of child vectors.
https://github.com/facebookincubator/velox/blob/2dc97deebba2664c4620f1e462476c5692691e4d/velox/exec/TableWriter.cpp#L129-L135

It looks to me more clean to not have this helper func makeRowVector() but directly call `std::make_shared<RowVector>` in the body of filterPartitionColumns().